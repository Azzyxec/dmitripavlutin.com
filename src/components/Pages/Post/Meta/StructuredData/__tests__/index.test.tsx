import * as React from 'react';
import { shallow } from 'enzyme';

import PostMetaStructuredData from '../index';

const post: Post = {
  description: 'Useful techniques to facilitate React server-side rendering.',
  html: '<div>JavaScript is a programming language</div>',
  modified: '2019-01-01',
  published: '2018-03-17',
  recommended: ['javascript-language'],
  slug: 'useful-techniques-react-server-side-rendering',
  tags: ['react', 'server side rendering'],
  thumbnail: {
    aspectRatio: 2,
    base64: 'base64',
    sizes: 'some sizes',
    src: '/source.png',
    srcSet: 'src-set',
  },
  title: 'Useful techniques to facilitate React server-side rendering',
};

const siteInfo = {
  title: 'Dmitri Pavlutin blog',
  description: 'Posts by Dmitri Pavlutin about software development',
  url: 'https://dmitripavlutin.com',
  repositoryUrl: 'https://github.com/panzerdp/dmitripavlutin.com',
};

const authorInfo = {
  name: 'Dmitri Pavlutin',
  description: 'Dmitri Pavlutin is a software developer specialized in Frontend technologies',
  profiles: {
    stackoverflow: 'https://stackoverflow.com/users/1894471/dmitri-pavlutin',
    twitter: 'https://twitter.com/panzerdp',
    linkedin: 'https://www.linkedin.com/in/dmitri-pavlutin/',
    github: 'https://github.com/panzerdp',
    facebook: 'https://www.facebook.com/dmitri.pavlutin',
  },
  nicknames: {
    twitter: 'panzerdp',
  },
};

const authorProfilePictureSrc = '/profile.png';

const props = {
  post,
  siteInfo,
  authorInfo,
  authorProfilePictureSrc,
};

describe('<PostMetaStructuredData />', function() {
  it('should render structured data', function() {
    const wrapper = shallow(<PostMetaStructuredData {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
