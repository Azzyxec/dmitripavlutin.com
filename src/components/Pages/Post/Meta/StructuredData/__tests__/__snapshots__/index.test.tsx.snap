// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostMetaStructuredData /> should render structured data 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    type="application/ld+json"
  >
    {
  "@context": "https://schema.org",
  "@type": "Article",
  "publisher": {
    "@type": "Organization",
    "name": "Dmitri Pavlutin blog",
    "logo": {
      "@type": "ImageObject",
      "url": "https://dmitripavlutin.com/profile.png"
    }
  },
  "author": {
    "@type": "Person",
    "name": "Dmitri Pavlutin",
    "image": {
      "@type": "ImageObject",
      "url": "https://dmitripavlutin.com/profile.png",
      "width": 256,
      "height": 256
    },
    "url": "https://dmitripavlutin.com",
    "sameAs": [
      "https://stackoverflow.com/users/1894471/dmitri-pavlutin",
      "https://twitter.com/panzerdp",
      "https://www.linkedin.com/in/dmitri-pavlutin/",
      "https://github.com/panzerdp",
      "https://www.facebook.com/dmitri.pavlutin"
    ],
    "description": "Software developer"
  },
  "headline": "Useful techniques to facilitate React server-side rendering",
  "url": "https://dmitripavlutin.com//useful-techniques-react-server-side-rendering/",
  "datePublished": "2018-03-17",
  "dateModified": "2019-01-01",
  "image": {
    "@type": "ImageObject",
    "url": "https://dmitripavlutin.com/source.png",
    "width": 720,
    "height": 400
  },
  "keywords": "react, server side rendering",
  "description": "Useful techniques to facilitate React server-side rendering.",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "https://dmitripavlutin.com"
  }
}
  </script>
</HelmetWrapper>
`;
